src/Analyzer.JsonRuleEngine/Expressions/AllOfExpression.cs

```diff
@@ -43,6 +43,13 @@ public override JsonRuleEvaluation Evaluate(IJsonPathResolver jsonScope)
                 foreach (var expression in AllOf)
                 {
                     var evaluation = expression.Evaluate(scope);
+
+                    // Filter out evaluation if it didn't find the scope to evaluate
+                    if (evaluation.NoScopesFound)
+                    {
+                        continue;
+                    }
+
                     evaluationPassed &= evaluation.Passed;
                     jsonRuleEvaluations.Add(evaluation);
```

Comment:
non-blocking: I generally use `continue` when there are large chunks of code I want to skip, or if a condition requiring code to be skipped is deeply nested and non-trivial to arrange differently.  Since this is only skipping two lines of code, I think it would read better having those two lines inside the `if`.

``` suggestion
                    // Only add this Evaluation if something was actually evaluated (Results are present)
                    if (evaluation.HasResults)
                    {
                        evaluationPassed &= evaluation.Passed;
                        jsonRuleEvaluations.Add(evaluation);
                    }
```

Reply:
I'm also wondering now - should the evaluation still be added to `jsonRuleEvaluations`, but just not be combined into `evaluationPassed`?  Maybe we should have a sync to figure out what we want the behavior to be.

Reply:
Added this as a note to discuss.